forc-pkg/src/pkg.rs

```diff
@@ -2799,8 +2797,26 @@ fn update_json_type_declaration(
     }
 }
 
-/// A `CompileResult` thats type is a tuple containing a `ParseProgram` and `Option<ty::TyProgram>`
-type ParseAndTypedPrograms = CompileResult<(ParseProgram, Option<ty::TyProgram>)>;
+/// Contains the lexed, parsed, and typed compilation stages of a program.
+pub struct Programs {
+    pub lexed: LexedProgram,
+    pub parsed: ParseProgram,
+    pub typed: Option<ty::TyProgram>,
+}
+
+impl Programs {
+    pub fn new(
+        lexed: LexedProgram,
+        parsed: ParseProgram,
+        typed: Option<ty::TyProgram>,
+    ) -> Programs {
+        Programs {
+            lexed,
+            parsed,
+            typed,
+        }
+    }
+}
 
 /// Compile the entire forc package and return the parse and typed programs
 /// of the dependancies and project.
```

Comment:
Should we also update this doc comment 👀

```suggestion
/// Compile the entire forc package and return the lexed, parsed and typed programs
/// of the dependancies and project.
```

Reply:
Good catch, i'll up date that comment now